staff/views/users.py

```diff
@@ -0,0 +1,133 @@
+import inspect
+
+from django.db import transaction
+from django.db.models import Q
+from django.db.models.functions import Lower
+from django.shortcuts import get_object_or_404, redirect
+from django.utils.decorators import method_decorator
+from django.views.generic import FormView, ListView, UpdateView
+
+from jobserver.authorization import roles
+from jobserver.authorization.decorators import require_permission
+from jobserver.authorization.utils import roles_for
+from jobserver.models import Backend, Org, User
+from jobserver.utils import raise_if_not_int
+
+from ..forms import UserForm, UserOrgsForm
+
+
+@method_decorator(require_permission("user_manage"), name="dispatch")
+class UserDetail(UpdateView):
+    form_class = UserForm
+    model = User
+    slug_field = "username"
+    slug_url_kwarg = "username"
+    template_name = "staff/user_detail.html"
+
+    @transaction.atomic()
+    def form_valid(self, form):
+        # remove the existing memberships so we can create new ones based on
+        # the form values
+        self.object.backend_memberships.all().delete()
+        for backend in form.cleaned_data["backends"]:
+            backend.memberships.create(user=self.object, created_by=self.request.user)
+
+        self.object.is_superuser = form.cleaned_data["is_superuser"]
+        self.object.roles = form.cleaned_data["roles"]
+        self.object.save()
+
+        return redirect(self.object.get_staff_url())
+
+    def get_context_data(self, **kwargs):
+        return super().get_context_data(**kwargs) | {
+            "orgs": self.object.orgs.order_by("name"),
+            "projects": self.object.projects.order_by("name"),
+        }
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+
+        # UserForm isn't a ModelForm so don't pass instance to it
```

Comment:
Should we be using `UpdateView` then?

Reply:
Yes, we get loads for free (I tried it without!), it's just that we don't want UserForm to be a model form (again it _could_ be, we're right in the grey area here).